### PR TITLE
Update license(s) of fixFormat to adapt license name

### PR DESCRIPTION
**Pull request by sufuf3**
Update the `tools/fixFormat.js` to fix format about `licenses` which can't adapt the license name.

It can't work when I run the `tools/fixFormat.js` to fix format about `licenses`, as the followings:
```
  "licenses": [
    "MIT",
    "GPLv2"
  ],
```
The result is same with the above.

I update the `tools/fixFormat.js` to fix format about `licenses` and let it can adapt the license name, as the followings:
```
  "licenses": [
    "MIT",
    "GPL-2.0"
  ],
```

### CODE REVIEW

**Comment by sufuf3** [comment]
@pvnr0082t I think the situation which you said is as the followings:
```
  "license": [
    "MIT",
    "GPLv2"
  ],
  "author": {
     "name": "jQuery Foundation and other contributors",
     "url": "https://github.com/jquery/jquery/blob/master/AUTHORS.txt"
  }
```
you use `license` not `licenses`, so it change the position of the licenses field.

But if you use `licenses: "MIT",` and run the origin `fixFormat.js`, you will see the position of the licenses field is changed as `license: "MIT",`

With this commit, I think I just want to update licenses of fixFormat to adapt license name.

File: tools/fixFormat.js [hunk 1]
```diff
@@ -103,9 +103,11 @@ function fixFormat() {
       }
     } else if (pkg.licenses !== undefined) {
       for (license in pkg.licenses) {
-        if (pkg.licenses[license].type !== undefined) {
```
Comment (petrgazarov) [inline tools/fixFormat.js]: What happens for license objects that have `type` property?

File: tools/fixFormat.js [hunk 2]
```diff
@@ -103,9 +103,11 @@ function fixFormat() {
...
-          if (licenses.indexOf(pkg.licenses[license].type) !== -1) {
-            pkg.licenses[license] = pkg.licenses[license].type;
+        if (pkg.licenses[license] !== undefined) {
+          if (licenses.indexOf(pkg.licenses[license]) !== -1) {
+            pkg.licenses[license] = pkg.licenses[license];
```
Comment (petrgazarov) [inline tools/fixFormat.js]: I don't think you need this line?

**Comment by PeterDaveHello** [comment]
@sufuf3 Can you also add test case like the others?

**Comment by sufuf3** [comment]
@petrgazarov Sorry about that I didn't know it is intentional. I just want it can work to let me add library smoothly.
And I saw the `licenses` part not use `function adapt(licenseName)`, so I add it.
But when I run the `fixFormat.js` it still not work.
When I try to remove the property  of `.type` and found it work.
So I add them.

I found that neither `pkg.license.type` or `pkg.licenses[license].type` always show its `undefined`, when it's a correct license name.

@PeterDaveHello I am not familiar with jasmine to write test case. If you could give me a try, I'll try my best to add it.

**Comment by petrgazarov** [comment]
@sufuf3 I see. It looks like the prior code checked license(s) objects, not license(s) strings.

That's hypothetically a licenses object:
```
licenses: {
  [
    {
      name: 'MIT license',
      type: 'MIT'
    },
    {
      name: 'GPL',
      type: 'GPL-2.0'
    }
  ]
}
```

I see most `package.json` have license properties as strings, so good catch!
I don't know if some license properties are still objects, but to provide backwards compatibility and support edge cases, I would suggesting doing all of the following:

- when `license` property exists and `license.type` exists, use `adapt` on the `type` property
- when `license` property exists and `license` is a string, use `adapt` on the string
- when `licenses` property exists, do the above two actions for each `license` in  `licenses` array.

**Comment by PeterDaveHello** [comment]
btw @petrgazarov take a look at pm when you have time plz, thx

**Comment by PeterDaveHello** [comment]
@mercyg if you'd be interested in code level contribution, consider to join this thread as a beginning could be good 👍

**Comment by sufuf3** [comment]
@petrgazarov Thnak you. I finally understand. I will improve this part of the code.

**Comment by sufuf3** [comment]
@petrgazarov I improve the code of fix license(s) part.
I test 4 situations as followings, and these situation is work:
1. `"license": "GPLv2",` --> `"license": "GPL-2.0",`
2. 
```js
"licenses": [
  {
    "name": "MIT license",
    "type": "MIT"
  },
  {
    "name": "GPL",
    "type": "GPLv2"
  }
],
```
-->
```js
"licenses": [
  "MIT",
  "GPL-2.0"
]
```
3. 
```js
"license": [
  {
    "name": "GPL",
    "type": "GPLv2"
  }
],
```
-->
```js
"license": "GPL-2.0"
```
4. 
```js
"licenses": [
  "MIT",
  "GPLv2"
]
```
-->
```js
"licenses": [
  "MIT",
  "GPL-2.0"
]
```
Please help me review this pull request. Thank you.

File: tools/fixFormat.js [hunk 3]
```diff
@@ -180,6 +174,24 @@ function fixFormat() {
...
   }
 
+  function adaptLicenseName(pkgName, licenseName, isRecognized) {
+    if (isRecognized == 1) {
+      if (licenses.indexOf(licenseName) !== -1) {
+        return licenseName;
+      } else if (adapt(licenseName)) {
+        return adapt(licenseName);
+      } else {
+        unRecognizedLicense(pkgName, licenseName);
```
Comment (petrgazarov) [inline tools/fixFormat.js]: Let's make this return `licenseName` in addition to logging the error. We don't want to assign `undefined` to `pkg.license` in `fixLicense` function.

**Comment by sufuf3** [comment]
It's a challenge for me to make code clear when log remove from `adaptLicenseName` function.
But as the above 4 situations these code can work.
I still learn how to write the test.
I will work on it ASAP.

**Comment by mercyg** [comment]
Hi @sufuf3,  is the test case written in Jasmine? If that is the case, I would like to help on that if you didn't start writing test case.

**Comment by sufuf3** [comment]
Yes, the test case is written in Jasmine.
I've been busy recently.
If you would like to help on this. I'll enormously grateful for your help.

If the code should improve let it pass the test, please let me know. I'll fix the code.

Thank you very much.

**Comment by mercyg** [comment]
@sufuf3 I don't know that much but I would like to help!

**Comment by mercyg** [comment]
@PeterDaveHello @sufuf3  I have tried to write a unit test in jasmine for the fixFromat. how do you run the jasmine in command line? I have tried running by typing 'jasmine' in the command line but that doesn't work

**Comment by petrgazarov** [comment]
There is a npm script you can execute. Check package.json file

Petr

**Comment by maruilian11** [comment]
I think it's a little wired if we use cases to fix the license field.
https://github.com/sufuf3/cdnjs/blob/6647418aa5015f68a5a03ce16b04ad92a3488da5/tools/fixFormat.js#L181
If we really need to do it in this way, we suppose to add every case as possible as we can rather than just for `GPL-2.0`, `GPL-3.0` and `Apache-2.0`.

I will recommend to warning the users to take a look at our [license list](https://github.com/cdnjs/cdnjs/blob/master/tools/license-list.json) and the **identifiers** in https://spdx.org/licenses/ once their license field is unrecognized by `fixFormat.js`.

Guys, how do you think? Thanks.

**Comment by sufuf3** [comment]
@maruilian11 according to your concern, I think that's why we need to add tests. (use `npm test`)
@petrgazarov sorry for late reply, I added the tests.
If I need to add or change something please let me know.
Many thanks.

**Comment by PeterDaveHello** [comment]
@sufuf3 don't you need to pass the CI first?

**Comment by sufuf3** [comment]
Yes, there is no error message from CI, so I push this PR again try to pass CI.
If there is still got CI test fault, I'll figure out.
Thanks.

**Comment by PeterDaveHello** [comment]
So did you pass the tests on local?

**Comment by sufuf3** [comment]
Yes, I passed `npm test`
Sorry for not mentioning it.

**Comment by sufuf3** [comment]
Sorry about that.
I am busy recently.
I'm not sure that I could improve this PR in my free time recently.
I will close this PR.
After I have free time to improve this PR, I would do it again, also reopen this PR again.

Sorry for bother you guys.
Many thanks.

**Comment by PeterDaveHello** [comment]
@sufuf3 I think we just need few more tasks to make this work?

**Comment by PeterDaveHello** [comment]
@sufuf3 rebase the branch and see if there is something you can do first, I'll help you dig the problem.

BTW, part of the codes could be refactored.

**Pull request closed without merging**